src/components/Trans.tsx

```diff
@@ -0,0 +1,130 @@
+import React from "react";
+
+import { useI18n } from "../i18n";
+
+const REGEXP = /({{\w+?}})|(<\w+?>)|(<\/\w+?>)/g;
+const KEY_REGEXP = /{{(\w+?)}}/;
+const TAG_START_REGEXP = /<(\w+?)>/;
+const TAG_END_REGEXP = /<\/(\w+?)>/;
+
+const getTransChildren = (
+  format: string,
+  props: {
+    [key: string]: React.ReactNode | ((el: React.ReactNode) => React.ReactNode);
+  },
+) => {
+  const stack: { name: string; children: React.ReactNode[] }[] = [
+    {
+      name: "",
+      children: [],
+    },
+  ];
+
+  format.split(REGEXP).forEach((match) => {
```

Comment:
the `split` here could be confusing as its introducing unncessary `undefined` tags so I think its better if we improve the split here so tokens are correct
eg for "Hello {{audience}}" its splitting to
```
 [ 'Hello ', '{{audience}}', undefined, undefined, '' ]
 ```
 so we should remove those `undefined` tokens

Reply:
I don't know if it's possible to write a regex with delimiters in JavaScript so that it won't produce `undefined` results. For cleaner code we could use `format.split(REGEXP).filter(Boolean)` which will filter our empty strings and `undefined`.

Reply:
yes I meant filtering only after tokens are generated